common/services/user-locations.js

```diff
@@ -0,0 +1,64 @@
+const axios = require('axios')
+
+const { AUTH_PROVIDERS, NOMIS_ELITE2_API } = require('../../config')
+const { decodeAccessToken } = require('../lib/access-token')
+const referenceDataService = require('./reference-data')
+
+function getUserLocations(token) {
+  return _getAgencies(token).then(agencies => {
+    return Promise.all(
+      agencies.map(agencyId =>
+        referenceDataService.getLocations({ nomisAgencyId: agencyId })
+      )
+    ).then(locations => locations.map(results => results[0].id))
+  })
+}
+
+function _getAgencies(token) {
+  let agenciesPromise
+
+  const tokenData = decodeAccessToken(token)
+
+  switch (tokenData.auth_source) {
+    case 'auth':
+      agenciesPromise = _getAuthAgencies(token)
+      break
+    case 'nomis':
+      agenciesPromise = _getNomisAgencies(token)
+      break
+  }
+
+  return agenciesPromise
+}
+
+function _getAuthAgencies(token) {
+  const tokenData = decodeAccessToken(token)
+
+  return axios
+    .get(AUTH_PROVIDERS.hmpps.groups_url(tokenData.user_name), {
+      headers: _getAuthHeaders(token),
+    })
+    .then(response => {
+      return response.data.map(group => group.groupCode.replace(/^PECS_/, ''))
+    })
+}
+
+function _getNomisAgencies(token) {
+  return axios
+    .get(NOMIS_ELITE2_API.user_caseloads_url, {
+      headers: _getAuthHeaders(token),
+    })
+    .then(response => {
```

Comment:
Does using lodash map not work here?

Could this work:

```js
.then(response => map(response.data, 'caseLoadId')
```

or maybe even chaining thens to make it easier to read:

```js
.then(response => response.data)
.then(data => map(data, 'caseLoadId'))
```